Blog: How to measure product-market fit (#6414)
PostHog/posthog.com: contents/blog/measure-product-market-fit.md

  - Product analytics
  - Product metrics
---

Startups can't succeed without [achieving product-market fit](/blog/product-market-fit-game) – it's one of the few things startup gurus agree on. It's often described as a feeling – a palpable excitement from users, or a sense you can't keep up with demand. You either have _it_, or you don't.

But product-market fit isn't just an ephemeral gut feeling. You can _measure_ it and _it moves_ as your customer's needs change. It's also a spectrum. Some products have rock solid product-market fit, others the green shoots of _potential_ fit. This guide will help you figure out where you are on that spectrum.

## First principles

Measuring product-market fit requires a combination of leading and lagging indicators:

**Leading indicators**, such as a surge in new users, suggest product-market fit.

**Lagging indicators**, such as those users returning repeatedly, confirm product-market fit.

Three metrics (two leading indicators and one lagging indicator) is a good starting point. You can add more if and when you need.

Why three? Because of what Brian Balfour, a serial co-founder and a former VP of Growth at Hubspot, calls The Trifecta:[^1]

> ![pmf trifecta](../images/blog/how-to-measure-product-market-fit/pmf-trifecta.png)

Author: ```suggestion
![pmf trifecta](../images/blog/how-to-measure-product-market-fit/pmf-trifecta.png)

Achieving all three is a reliable sign of product-market fit, but it rarely comes easily.

```

| <span className="text-green text-lg">✔</span> Can be broken down by user cohorts for comparison                          |           |
| <span className="text-green text-lg">✔</span> You can A/B test ways to improve user engagement                           |           |

## Indicator #4: Paying ICP customers

- **Type:** Leading indicator
- **Good for:** B2B SaaS products and subscription services

It's tempting to treat revenue as a measure of product-market fit, but it's imperfect. It's easily skewed by one or two big contracts, selling promises, or "design partners" who guide your product to solve their very specific needs.

Instead, create an [Ideal Customer Profile](/blog/creating-ideal-customer-profile) (ICP) and track growth among paying customers who fit your ICP. Your ICP should be specific, and include needs, haves, and common attributes. 

Here's PostHog's first ICP:

![posthog icp](../images/blog/pmf-game-guide/posthog-icp.png)

> How much your ICP customers are paying is _less important_ than _how many_ ICP customers you have, and if they're growing in number.
> 
> Consistent growth in ICP customers who pay is a strong sign of product-market fit. It means you have a product they want, and you can acquire similar customers with ease.

Author: ```suggestion
_How much_ your ICP customers are paying is less important than _how many_ ICP customers you have, and if they're growing in number. Can you onboard 5 paying ICP customers, then 10, 20, 50, and so on?

Consistent growth in ICP customers who pay is a strong sign of product-market fit. It means you have a product they want, and you can acquire similar customers with ease.
```

Author: Does this work better @ivanagas?

Reviewer: yup

### Who should use the PMF Survey?

Feedback is essential to any startup, so it's more a question of how you collect it. Some teams may find an ad hoc approach suits their style. That's fine – we haven't used the PMF Survey at PostHog, for example, and we have strong product-market fit.

But, as Superhuman proves, the survey is a powerful way to systematize finding product-market fit. It's also useful for tracking whether it's improving as you ship new features, making it useful both before _and_ after you have it.

| **Pros**                                                                                                                 | **Cons**                                                                                                                 |
|--------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------| 

> | <span className="text-green text-lg">✔</span> Can help guide product development in the right direction                  | <span className="text-red text-lg">✖</span> PMF Survey is somewhat labor-intensive                                       |

Author: Heh, I wrote this when Surveys had fewer features!

Maybe "process-intensive" is better? The point I'm trying to make is it's a very defined, repetitive process, which might not vibe with a team who wants to move faster etc.

> | <span className="text-green text-lg">✔</span> Can help guide product development in the right direction                  | <span className="text-red text-lg">✖</span> PMF Survey is somewhat labor-intensive                                       |

Author: Maybe I can make this something like... could slow you down?

> | <span className="text-green text-lg">✔</span> Can help guide product development in the right direction                  | <span className="text-red text-lg">✖</span> PMF Survey is somewhat labor-intensive                                       |

Author: Will probably just remove this.

## Indicator #3: User engagement

- **Type:** Leading indicator
- **Good for:** Any software product

Are users experiencing the _real value_ of your product? Spoiler: logging in ≠ engagement.

You want to see user engagement growing faster than, or in line with, new users. Engagement growing faster than signups is a strong predictor of product-market fit.

![good pmf](../images/blog/how-to-measure-product-market-fit/good-pmf.png)

Growing signups with flat user engagement is flat suggests you _don't_ have product-market fit – most likely people are trying your product, but not coming back.

![bad pmf](../images/blog/how-to-measure-product-market-fit/bad-pmf.png)

At PostHog, we created a user engagement metric we call Discoveries. We define a discovery as:

- **Analyzing any insight** – Viewing an insight for 10 seconds or more.
- **Analyzing a recording** – Watching a recording for 10 seconds or more.
- **Analyzing a correlation** – Viewing a correlation for 10 seconds or more.
- **Analyzing a dashboard** – Viewing a dashboard for 10 seconds or more.

We also track things like people inviting new team members, which we found correlates strongly with retention. 

> If you find your chosen engagement metric doesn't correlate to retention you either:

Reviewer: ```suggestion
If you find your chosen engagement metric doesn't correlate to retention, you either:
```

1. Don't have product-market fit.
2. Are tracking the wrong metric(s).

> > 💡 **PostHog Tip:** You can create your own user engagement metric in PostHog by creating an [Action](/docs/data/actions). Actions enable you to combine individual events (e.g. editing a page, sending a message, etc.) and track them as one metric. You can use Actions in numerous insight types, including trends, funnels, and retention insights.

Reviewer: ```suggestion
> 💡 **PostHog Tip:** You can create your own user engagement metric in PostHog by creating an [action](/docs/data/actions). Actions enable you to combine individual events (e.g. editing a page, sending a message, etc.) and track them as one metric. You can use actions in numerous insight types, including trends, funnels, and retention insights.
```

| <span className="text-green text-lg">✔</span> Tracks the real value users gain from your product                         | <span className="text-red text-lg">✖</span> High engagement isn't necessarily predictive of revenue                      |
| <span className="text-green text-lg">✔</span> Passive to maintain – set it up once and track forever                     | <span className="text-red text-lg">✖</span> Can provide a false positive if you're tracking the wrong things             |
| <span className="text-green text-lg">✔</span> Easy to understand for all teams                                           | <span className="text-red text-lg">✖</span> Requires user feedback or session replays to understand behavior             |
| <span className="text-green text-lg">✔</span> Can be broken down by user cohorts for comparison                          |           |
| <span className="text-green text-lg">✔</span> You can A/B test ways to improve user engagement                           |           |

## Indicator #4: Paying ICP customers

- **Type:** Leading indicator
- **Good for:** B2B SaaS products and subscription services

> It's tempting to treat revenue as a measure of product-market fit, but it's imperfect. It's easily skewed by one or two big contracts, or "design partners" who guide your product to solve their very specific needs.

Reviewer: ```suggestion
It's tempting to treat revenue as a measure of product-market fit, but it's imperfect. It's easily skewed by one or two big contracts, selling promises, or "design partners" who guide your product to solve their very specific needs.
```

Instead, create an [Ideal Customer Profile](/blog/creating-ideal-customer-profile) (ICP) and track growth among paying customers who fit your ICP. Your ICP should be specific, and include needs, haves, and common attributes. 

Here's PostHog's first ICP:

![posthog icp](../images/blog/pmf-game-guide/posthog-icp.png)

> How much your ICP customers are paying is _less important_ than _how many_ ICP customers you have, and if they're growing in number.

Reviewer: I don't fully buy this argument. If you have a couple big customers in your ICP with long sales cycles, could still be a good sign?

Reviewer 2: Agree with Ian here. It's very case by case and there'd be many caveats to explain it fully. Best to just say that teams need to form their own opinion on this balance IMHO.

Consistent growth in ICP customers who pay is a strong sign of product-market fit. It means you have a product they want, and you can acquire similar customers with ease.

> [I WILL ADD A GRAPHIC HERE WITH SOME LINES ON IT, PROBABLY]

Reviewer: 🤔

> [I WILL ADD A GRAPHIC HERE WITH SOME LINES ON IT, PROBABLY]

Author: ```suggestion
```

| <span className="text-green text-lg">✔</span> Defining an ICP is an essential step toward product-market fit             | <span className="text-red text-lg">✖</span> Not useful for ad-supported products (e.g. social networks)                  |
| <span className="text-green text-lg">✔</span> Focuses on your most valuable users                                        | <span className="text-red text-lg">✖</span> Not useful when you're pre-revenue                                           |
| <span className="text-green text-lg">✔</span> Easy to understand – growth good, no growth bad!                           | <span className="text-red text-lg">✖</span> Hard to track ICP automatically                                  |
| <span className="text-green text-lg">✔</span> Won't tell you why you do or don't have product-market fit                 |                                 |

## Indicator #5: Retention curve

- **Type:** Lagging indicator
- **Good for:** Any software product

> Retention is _the main_lagging indicator of product-market fit for any type of product. If your retention curve fattens _at any point_, it's a strong sign you have product-market fit.

Reviewer: ```suggestion
Retention is _the main_lagging indicator of product-market fit for any type of product. If your retention curve flattens _at any point_, it's a strong sign you have product-market fit.
```

![retention curve](../images/blog/how-to-measure-product-market-fit/retention-curve.png)

You'll want to track active users over a long(ish) period of time – weeks or months, rather than days. It's best to track events or actions where a user is experiencing the _real value_ of your product. Again, logging in does not count!

For our product PostHog, an [all-in-one analytics platform](http://posthog.com/), this might look like:

> Show `unique users` who performed a `discovery` for the first time in the last `12 months` and then came back to perform a `discovery` on any of the next months.

It's also a good idea to track different cohorts, such as retention for users who match your ideal customer profile. This will reveal how strong your fit is for different users, and plan your roadmap accordingly.

> 💡 **PostHog Tip:** It's easy to track retention in PostHog using the [retention insight type](/docs/product-analytics/retention), which you can segment by cohorts (e.g. users by job title, age, etc.). See also: [How to discover features that drive user retention](/tutorials/feature-retention).

### Who should use retention?

> Everyone. It's easy to understand and a reliable way to confirm product-market fit.

Reviewer: What about a product that is only used a few times a year, like Airbnb or an online car dealership? Retention would be hard to track and not super useful.

Author: Could probably downgrade to almost everyone, but in both cases they'd be looking to retain properties and dealers, right? I can add some examples of what retention looks like for different types of products here.

| <span className="text-green text-lg">✔</span> Reliable lagging indicator of product-market fit                           | <span className="text-red text-lg">✖</span> Not predictive of revenue – will users actually pay for your product?        |
| <span className="text-green text-lg">✔</span> Can be broken down by cohort for comparison                                | <span className="text-red text-lg">✖</span> You need several weeks or months of data to get a reliable signal            |
| <span className="text-green text-lg">✔</span> Easy to understand – flat is good, not flat is bad!                        | <span className="text-red text-lg">✖</span> Won't help you understand why you do or don't have product-market fit        |

## Indicator #6: The burn multiple

- **Type:** Lagging indicator
- **Good for:** Late-stage companies seeking investment

Created by former PayPal COO David Sacks, burn multiple measures how much you're spending to grow revenue. The less you spend, the more efficient your growth. More efficient growth = stronger product-market fit.

> Because it measures efficiency rather than product-market fit directly, burn multiple is really a way to measure how _strong_ your product-market fit is. It's useful for late-stage startups seeking further investment, but it's a board-level metric – i.e. product teams won't find it useful.

Reviewer: nice

### Who should use burn multiple?

It's a good metric to add once you believe you have the trifecta: consistent growth, real user engagement, and strong retention. It'll help you understand how investors will evaluate your product-market fit.

| **Pros**                                                                                                                 | **Cons**                                                                                                                 |
|--------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------| 
| <span className="text-green text-lg">✔</span> It's a cumulative metric that's influenced by all parts of your startup    | <span className="text-red text-lg">✖</span> It's a significantly lagging indicator                                       |

> | <span className="text-green text-lg">✔</span> It's easy to calculate – it's just two numbers any founder should have     | <span className="text-red text-lg">✖</span> It's not useful if you're pre-revenue                                        |

Reviewer: ```suggestion
| <span className="text-green text-lg">✔</span> It's easy to calculate – it's just two numbers any founder should have     | <span className="text-red text-lg">✖</span> It's not useful if you're pre-revenue or even early revenue                                      |
```

| <span className="text-green text-lg">✔</span> Useful for sales-led companies                                             |                              |

## Summing up

How you measure product-market fit depends on a few factors, such as:

1. The stage of your company – are you pre- or post-revenue?
2. The type of product you're building – SaaS, marketplace etc.
3. Your ideal customers – are they consumers, SMBs, or enterprises?
4. How you generate / plan to generate revenue – are you self-serve, or sales-led?

For most early-stage startups, word-of-mouth growth, user engagement, and retention is the place to start. That's the traditional trifecta.

For any B2B product, it's important to add paying ICP customers as soon as you're ready to generate revenue. Breakdown retention by user cohorts to verify your ICP customers retain better than non-ICP customers. If they do, and your ICP customers are growing, it's a good sign you're on the right path.

The PMF Survey is best used to:

1. Figure out why you do or don't have product-market fit
2. To improve the strength of your product-market fit 
3. To track how strong your product-market fit is over time

> But remember: talking to users and gathering feedback remains essential whether you use the survey, or not.

Reviewer: ```suggestion
But remember: [talking to users](/blog/10x-engineers-do-user-interviews) and gathering feedback remains essential whether you use the survey, or not.
```

- **Good for:** Product-led companies

You _can't_ validate product-market fit _using word of mouth alone_. That way lies madness. But it is a useful indicator _when confirmed by other metrics_, such as user engagement.

Hopefully, you already have a strong grasp of your organic user growth via user signups, or whatever metric makes sense for your product. But you can, with a little work, also track word-of-mouth awareness and sentiment of your product. Here are a few options:

### 1. Brand mention alerts

Use a tool like [Syften](https://syften.com/) to monitor social media, communities, and newsletters for people talking about you. We use it at PostHog to send alerts to a `#brand-alerts` channel in our Slack. 

Tracking brand mentions is more about understanding what people are saying about you, rather than tracking a trend. People spontaneously sharing their love for your product is a good sign. Identify who these people are and why your product is such a good fit for them.

> **❗️Important:** It's _much easier_ to track mentions if your company has a unique name. You're bang out of luck if it's a common noun – e.g. Apple, Amplitude. 

### 2. Searches for your brand

> **Google Trends** is a good option if you have a unique brand name. Just plug your brand into Trends to see how searches for your brand are trending over time.

Reviewer 2: ```suggestion
**[Google Trends](https://trends.google.com/trends/)** is a good option if you have a unique brand name. Just plug your brand into Trends to see how searches for your brand are trending over time.
```

![google trend comparison](../images/blog/how-to-measure-product-market-fit/posthog-with-logrocket.png)

At PostHog, we also use [Glimpse](https://meetglimpse.com/) – an extension that augments Trends by adding trend lines, forecasts, and real search volumes. 

![trends with Glimpse](../images/blog/how-to-measure-product-market-fit/glimpse.png)

The above shows how PostHog showed signs of product-market fit in late 2021 / early 2022, which grew stronger from mid-2022 onwards. While exponential growth is desirable, linear growth is still a good sign.

Glimpse also suggests similar companies – useful for comparing your growth with those you know have product-market fit.

![related companies](../images/blog/how-to-measure-product-market-fit/related-companies.png)

### 3. Organic traffic to your homepage

You can verify the above using organic visitors to your website homepage as a proxy for word of mouth. We track this in PostHog as an insight configured to show:

1. Unique users who visited the homepage via a search engine.
2. Unique users who visited the homepage directly.
3. While excluding users with an identified email address.

> This captures new users who visit our homepage by searching for us, typing in our website address, or via "dark social" sources like Slack and WhatsApp.

Reviewer 2: ```suggestion
This captures new users who visit our homepage by searching for us, typing in our website address, or via dark social sources like Slack and WhatsApp.
```
It's a nit, but I don't generally like quotations like this as they can come across as snarky.

![retention curve](../images/blog/how-to-measure-product-market-fit/retention-curve.png)

You'll want to track active users over a long(ish) period of time – weeks or months, rather than days. It's best to track events or actions where a user is experiencing the _real value_ of your product. Again, logging in does not count!

For our product PostHog, an [all-in-one analytics platform](http://posthog.com/), this might look like:

> Show `unique users` who performed a `discovery` for the first time in the last `12 months` and then came back to perform a `discovery` on any of the next months.

It's also a good idea to track different cohorts, such as retention for users who match your ideal customer profile. This will reveal how strong your fit is for different users, and plan your roadmap accordingly.

> 💡 **PostHog Tip:** It's easy to track retention in PostHog using the [retention insight type](/docs/product-analytics/retention), which you can segment by cohorts (e.g. users by job title, age, etc.). See also: [How to discover features that drive user retention](/tutorials/feature-retention).

### Who should use retention?

> Everyone. It's easy to understand and a reliable way to confirm product-market fit.

Author: ```suggestion
Nearly everyone. It's easy to understand and a reliable way to confirm product-market fit. Just remember to consider what "retention" means for your product. 

As a B2B SaaS product, we care about retaining organizations and individual users. If individual users keep coming back, we're confident their respective orgs will retain well, too. A marketplace like Airbnb, meanwhile, will want to track retention for:

1. Owners – do owners keep their properties listed on Airbnb?
2. End users – do customers return to book more stays?
```

Author: @ivanagas Added some detail here

| <span className="text-green text-lg">✔</span> A steady decline is a strong indicator of product-market fit               | <span className="text-red text-lg">✖</span> It won't help you understand why you do or don't have product-market fit     |
| <span className="text-green text-lg">✔</span> It's useful for investors                                                  | <span className="text-red text-lg">✖</span> Skewed by non-product factors – e.g. poor cost control                       |
| <span className="text-green text-lg">✔</span> It's easy to understand                                                    |        |
| <span className="text-green text-lg">✔</span> Useful for mid to late-stage startups                                      |        |

## Indicator #7: CLV/CAC

- **Type:** Lagging indicator
- **Good for:** Late-stage, sales-led startups

> **CLV** = customer lifetime value – i.e. the total value of a customer after churn and the cost of supporting / retaining them.
>
> **CAC** = customer acquisition cost - i.e. total sales and marketing spend divided by how many customers you acquired.

> CLV/CAC is a more targeted way to track "efficiency" than burn multiple. It's a ratio calculated by dividing your customer lifetime value[^6] (CLV) by your customer acquisition cost (CAC). 

Reviewer 2: ```suggestion
CLV/CAC is a more targeted way to track efficiency than burn multiple. It's a ratio calculated by dividing your customer lifetime value[^6] (CLV) by your customer acquisition cost (CAC). 
```

Reviewer 2: Also, you should define acronyms the first time they are used, not the second.

> The ratio reveals how efficiently you're acquiring customers. A ratio of 1x means your CLV and CAC are the same – i.e. you're only earning enough to cover your acquisition costs. A ratio of 2x means you're earning twice as much as you're spending to acquire new customers. 

Reviewer 2: ```suggestion
The ratio reveals how efficiently you're acquiring customers. A ratio of 1:1 means your CLV and CAC are the same – i.e. you're only earning enough to cover your acquisition costs. A ratio of 2:1 means you're earning twice as much as you're spending to acquire new customers. 
```

Changed these to ratios

A CLV/CAC of 3x and a CAC payback of less than 18 months ideal[^7] for sales-lead products.

> | **CLV/CAC ratio** | **Product-market fit** |

Reviewer 2: These need to be ratios too
